x-pack/plugins/security_solution/public/common/components/visualization_actions/use_lens_attributes.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useParams, useLocation } from 'react-router-dom';
+import { SecurityPageName } from '../../../../common/constants';
+import { NetworkRouteType } from '../../../network/pages/navigation/types';
+import { useSourcererDataView } from '../../containers/sourcerer';
+import { useDeepEqualSelector } from '../../hooks/use_selector';
+import { inputsSelectors } from '../../store';
+import { LensAttributes, GetLensAttributes } from './types';
+import {
+  getHostDetailsPageFilter,
+  filterNetworkExternalAlertData,
+  filterHostExternalAlertData,
+  getIndexFilters,
+} from './utils';
+
+export const useLensAttributes = ({
+  lensAttributes,
+  getLensAttributes,
+  stackByField,
+}: {
+  lensAttributes?: LensAttributes | null;
+  getLensAttributes?: GetLensAttributes;
+  stackByField?: string;
+}): LensAttributes | null => {
+  const { selectedPatterns, dataViewId } = useSourcererDataView();
+  const getGlobalQuerySelector = useMemo(() => inputsSelectors.globalQuerySelector(), []);
+  const getGlobalFiltersQuerySelector = useMemo(
+    () => inputsSelectors.globalFiltersQuerySelector(),
+    []
+  );
+  const query = useDeepEqualSelector(getGlobalQuerySelector);
+  const filters = useDeepEqualSelector(getGlobalFiltersQuerySelector);
+  const { detailName, tabName } = useParams<{ detailName: string | undefined; tabName: string }>();
+  const location = useLocation();
```

Comment:
```suggestion
  const [ { detailName, pageName, tabName }] = useRouteSpy();
```
Get these from `useRouteSpy` instead plz